spec/controllers/api/v4/questionnaire_responses_controller_spec.rb

```diff
@@ -0,0 +1,235 @@
+require "rails_helper"
+
+RSpec.describe Api::V4::QuestionnaireResponsesController, type: :controller do
+  let(:request_user) { create(:user) }
+  let(:request_facility_group) { request_user.facility.facility_group }
+  let(:request_facility) { create(:facility, facility_group: request_facility_group) }
+  let(:model) { QuestionnaireResponse }
+  let(:build_payload) { -> { build_questionnaire_response_payload } }
+  let(:build_invalid_payload) { -> { build_invalid_questionnaire_response_payload } }
+
+  before do
+    @questionnaire_types = stub_questionnaire_types
+  end
+
+  def create_record(options = {})
+    facility = options[:facility] || request_facility
+    questionnaire = options[:questionnaire] || create(:questionnaire)
+    create(:questionnaire_response, questionnaire: questionnaire, facility: facility)
+  end
+
+  def create_record_list(n, options = {})
+    facility = options[:facility] || request_facility
+    questionnaire = options[:questionnaire] || create(:questionnaire)
+
+    create_list(:questionnaire_response, n, questionnaire: questionnaire, facility: facility)
+  end
+
+  it_behaves_like "a sync controller that authenticates user requests"
+  it_behaves_like "a sync controller that audits the data access: sync_to_user"
+
+  describe "a sync controller that audits the data access: sync_from_user" do
```

Comment:
We aren't using the `a sync controller that audits the data access: sync_from_user` shared spec because of these subtle differences:
- Since `content` can be merged from `old` requests, it is technically an `update`. This is different from other resources where `old` requests are disregarded.
- The facility ID of a questionnaire cannot be changed once set, and the spec needs to accommodate for that